akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -698,6 +701,24 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
     }
   }
 
+  // set the future that completes when the current stream for a given name completes
+  @tailrec
+  private def updateStreamCompletion(streamName: String, streamCompleted: Future[Done]): Unit = {
```

Comment:
Yeah, maybe we could just add _all_ of them in a set, the restarted ones will have completed as well in the end, no?